code/modules/multiz/movement.dm

```diff
@@ -140,14 +162,39 @@
 
 	if(throwing)
 		return
+	if(ismob(src))
+		var/mob/H = src //VOREStation Edit Start. Flight on mobs.
+		if(H.flying == 1) //Some other checks are done in the wings_toggle proc
+			if(H.nutrition > 2)
+				H.nutrition -= 2 //You use up 2 nutrition per TILE and tick of flying above open spaces. If people wanna flap their wings in the hallways, shouldn't penalize them for it.
+			if(H.incapacitated(INCAPACITATION_ALL))
+				H.stop_flying()
+				//Just here to see if the person is KO'd, stunned, etc. If so, it'll move onto can_fall.
+			else if (H.nutrition > 1000) //Eat too much while flying? Get fat and fall.
+				to_chat(H, "<span class='danger'>You're too heavy! Your wings give out and you plummit to the ground!</span>")
+				H.stop_flying() //womp womp.
+			else if(H.nutrition < 300 && H.nutrition > 289) //290 would be risky, as metabolism could mess it up. Let's do 289.
+				to_chat(H, "<span class='danger'>You are starting to get fatigued... You probably have a good minute left in the air, if that. Even less if you continue to fly around! You should get to the ground soon!</span>") //Ticks are, on average, 3 seconds. So this would most likely be 90 seconds, but lets just say 60.
+				H.nutrition -= 10
+				return
+			else if(H.nutrition < 100 && H.nutrition > 89)
+				to_chat(H, "<span class='danger'>You're seriously fatigued! You need to get to the ground immediately and eat before you fall!</span>")
+				return
+			else if(H.nutrition < 2) //Should have listened to the warnings!
+				to_chat(H, "<span class='danger'>You lack the strength to keep yourself up in the air...</span>")
+				H.stop_flying()
+			else
+				return
+		if(H.grabbed_by.len) //If you're grabbed (presumably by someone flying) let's not have you fall. This also allows people to grab onto you while you jump over a railing to prevent you from falling!
```

Comment:
Can people grab each other at the same time? If so, this might allow people to walk across the void by holding onto each other.

Reply:
Hmm, that is problematic. Didn't think of that. Going to test some potential fixes for this. Possibly doing a check "if var/obj/item/weapon/grab/G in contents? Will fiddle with it.